lib/Sub/Override.pm

```diff
@@ -23,119 +23,118 @@ sub new {
 }
 
 sub replace {
-    my ( $self, $sub_to_replace, $new_sub ) = @_;
-    $sub_to_replace = $self->_get_fully_qualified_sub_name($sub_to_replace);
-    $self->_ensure_code_slot_exists($sub_to_replace)->_validate_sub_ref($new_sub);
-    {
-        no strict 'refs';
-        $self->{$sub_to_replace} ||= *$sub_to_replace{CODE};
-        no warnings 'redefine';
-        *$sub_to_replace = $new_sub;
+    my ( $self, %subs ) = @_;
+    while (my ($sub_name, $new_sub_ref) = each %subs) {
```

Comment:
Annoyingly, I can't recall the exact details now (had something to do with wrapping subs), but we did something very similar to this for a client project. There were cases in which we found that the order was important, so we guaranteed the order or replacement/wrapping would be the order in which they were declared:

```perl
sub wrap ($self, @wrappers) {
    if ( @wrappers % 2 ) {
        croak("Arguments to wrap() must be an even-sized list of k/v pairs");
    }
    while (@wrappers) {
        my ( $subname, $subref ) = splice @wrappers, 0, 2;
        ...
```

Whether you feel this change is worthwhile is up to you.

Reply:
I seem to recall that this had something to do with "stacked" wrappers and the order of operations sometimes causing issues if they were not deterministic.

Reply:
I've been mulling this over and haven't been able to see so far how this would be a problem in Sub::Override code. Currently you can't wrap more than once, 'wrap' behaves like 'replace' where if you wrap the same routine a second time, the original wrap disappears and the new wrap takes over. I can see that a stacked-type wrap would likely be more desirable (a-la moose's 'around'), but that would be a new feature for this.